### PR TITLE
Use rev-parse --show-toplevel

### PR DESCRIPTION
**Pull request by asottile**
A first step towards #213

### CODE REVIEW

**Comment by asottile** [comment]
I have a sneaking suspicion that this broke something at some point, but I can't remember why...

https://github.com/pre-commit/pre-commit/commit/bf457349b3a1ca16c47b6878c05c9a3c11ee712d

**Comment by asottile** [comment]
Actually, I bet that was plumbum's fault and I just didn't know it at the time

File: testing/util.py [hunk 1]
```diff
@@ -41,6 +41,14 @@ def get_head_sha(dir):
         return cmd_output('git', 'rev-parse', 'HEAD')[1].strip()
 
 
+def normpath2(path):
+    """Like os.path.normpath, but does lower() for windows.  This is obsolete
+    in python3 which does this already.
+    """
+    path = os.path.normpath(path)
+    return path.lower() if os.name == 'nt' else path
```
Comment (chriskuehl) [inline testing/util.py]: does `normcase` do the right thing?
https://docs.python.org/2/library/os.path.html#os.path.normcase

although...

> On Unix and Mac OS X, this returns the path unchanged; on case-insensitive filesystems, it converts the path to lowercase.

this seems a little dubious, doesn't OS X use a case-insensitive fs by default?

**Comment by chriskuehl** [comment]
since you're in the midwest



shipit

File: testing/util.py (hunk 1 above)
Comment (asottile) [inline testing/util.py]: oh yeah I could probably use normcase, osx should be doing the same though!

File: pre_commit/git.py [hunk 2]
```diff
@@ -15,16 +16,12 @@
...
-            path = os.path.normpath(os.path.join(path, '../'))
-    raise FatalError(
-        'Called from outside of the gits. '
-        'Please cd to a git repository.'
-    )
+    try:
+        return cmd_output('git', 'rev-parse', '--show-toplevel')[1].strip()
+    except CalledProcessError:
+        raise FatalError(
+            'Called from outside of the gits.  Please cd to a git repository.'
```
Comment (chriskuehl) [inline pre_commit/git.py]: two spaces here intentional?

**Comment by chriskuehl** [comment]
otherwise shipit

File: pre_commit/git.py (hunk 2 above)
Comment (asottile) [inline pre_commit/git.py]: yeah

**Pull request merged**